### DISCUSSION TITLE
make command

### DISCUSSION THREAD

**Discussion by V-Deepak-7** [The-OpenROAD-Project/OpenROAD-flow-scripts#2080] [Q&A]
Hello all,
This might sound stupid. I am new to this make scripts and software integration of the hardware design. 
I learned by using the make -j <threads> we speed up the work by using more threads essentially but in this case it is still taking the same amount of time as the normal make. Does it has something to do with the how the tcl scripts that are written in the backend to run the openroad and yosys? I would like to speed up the process any suggestions?

**Comment by maliberty** [comment]
The openroad steps are serial and there is not much place for parallel work so -j won't help much.  If you are in a prototyping mode there are some short cuts but for production runs there is no easy speedup.

**Reply by V-Deepak-7** [reply]
if that is the case in openroad documentation I have seen the following commands {openroad [-help] [-version] [-no_init] [-exit] [-gui]
         [-threads count|max] [-log file_name] cmd_file} the "-threads count|max" what does this command do? If this leverages the threads usage how to use it in the ORFS?

**Reply by maliberty** [reply]
Within one invocation of OR we may internally use threads to speed up one step.  ORFS automatically sets NUM_CORES to the avaiable cpus.

**Reply by V-Deepak-7** [reply]
I highly appreciate your swift responses @maliberty . I am currently trying to run a systolic array design on ORFS, which has been running for over 32 hours. Is this common based on the design complexity? I am working on a project where I have multiple designs to be evaluated. Is there any quick way to do this? I was thinking that instead of the ORFS, just writing the scripts to Yosys and OR would be a better solution, or would it still take the same amount of time? Can you provide any link to the manual of OR and yosys? that would be very helpful for me. Thank you in advance for responding.

**Reply by maliberty** [reply]
How large is the array in instances?  Are the elements at gate level or are you using hierarchy?  The runtime is dependent on many factors so more info is needed.
